integrations/mcp/src/haystack_integrations/tools/mcp/mcp_toolset.py

```diff
@@ -0,0 +1,232 @@
+
+    # Create the toolset - this will automatically discover all available tools
+    # You can optionally specify which tools to include
+    mcp_toolset = MCPToolset(
+        server_info=server_info,
+        tool_names=["get_current_time"]  # Only include the get_current_time tool
+    )
+
+    # Create a pipeline with the toolset
+    pipeline = Pipeline()
+    pipeline.add_component("llm", OpenAIChatGenerator(model="gpt-4o-mini", tools=mcp_toolset))
+    pipeline.add_component("tool_invoker", ToolInvoker(tools=mcp_toolset))
+    pipeline.add_component(
+        "adapter",
+        OutputAdapter(
+            template="{{ initial_msg + initial_tool_messages + tool_messages }}",
+            output_type=list[ChatMessage],
+            unsafe=True,
+        ),
+    )
+    pipeline.add_component("response_llm", OpenAIChatGenerator(model="gpt-4o-mini"))
+    pipeline.connect("llm.replies", "tool_invoker.messages")
+    pipeline.connect("llm.replies", "adapter.initial_tool_messages")
+    pipeline.connect("tool_invoker.tool_messages", "adapter.tool_messages")
+    pipeline.connect("adapter.output", "response_llm.messages")
+
+    # Run the pipeline with a user question
+    user_input = "What is the time in New York? Be brief."
+    user_input_msg = ChatMessage.from_user(text=user_input)
+
+    result = pipeline.run({"llm": {"messages": [user_input_msg]}, "adapter": {"initial_msg": [user_input_msg]}})
+    print(result["response_llm"]["replies"][0].text)
+    ```
+
+    You can also use the toolset via MCP SSE to talk to remote servers:
+    ```python
+    from haystack_integrations.tools.mcp import MCPToolset, SSEServerInfo
+    from haystack.components.tools import ToolInvoker
+
+    # Create the toolset with an SSE connection
+    sse_toolset = MCPToolset(
+        server_info=SSEServerInfo(base_url="http://some-remote-server.com:8000"),
+        tool_names=["add", "subtract"]  # Only include specific tools
+    )
+
+    # Use the toolset as shown in the pipeline example above
+    ```
+    """
+
+    def __init__(
+        self,
+        server_info: MCPServerInfo,
+        tool_names: list[str] | None = None,
+        connection_timeout: int = 30,
+        invocation_timeout: int = 30,
+    ):
+        """
+        Initialize the MCP toolset.
+
+        :param server_info: Connection information for the MCP server
+        :param tool_names: Optional list of tool names to include. If provided, only tools with
+                          matching names will be added to the toolset.
+        :param connection_timeout: Timeout in seconds for server connection
+        :param invocation_timeout: Default timeout in seconds for tool invocations
+        :raises MCPToolNotFoundError: If any of the specified tool names are not found on the server
+        """
+        # Initialize with empty tools list first
+        super().__init__(tools=[])
```

Comment:
Small nit, why not super at the end once we have all the tools set up instead of using `self.add`?